cpp/src/arrow/sparse_tensor_format.h

```diff
@@ -0,0 +1,25 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#ifndef ARROW_SPARSE_TENSOR_FORMAT_H
+#define ARROW_SPARSE_TENSOR_FORMAT_H
+
+struct SparseTensorFormat {
+  enum type { COO, CSR };
```

Comment:
Is this worth its own file?

Reply:
`SparseTensorFormat::type` is referred from metadata-internal.h and I didn't want to include sparse_tensor.h in metadata-internal.h, so I needed to extract `SparseTensorFormat` from sparse_tensor.h.

Reply:
Since `metadata-internal.h` is an internal header that is only used during compilation, I don't think it is so bad to include `arrow/sparse_tensor.h`. Is it OK to combine this file into sparse_tensor.h?